src/battle_game/core/bullets/crossing_obstacle_cannonball.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include "battle_game/core/bullet.h"
+
+namespace battle_game::bullet {
+class crossing_obstacles_cannonball : public Bullet {
```

Comment:
Should use class name like `CrossingObstaclesCannonball` 
(And your class name may be a little bit long, which doesn't matter a lot though)